### PR TITLE
Feature/create admin and client controller for questions

### PR DESCRIPTION
**Pull request by bogdanrymar**
I created client and admin parts of QuestionsController and add some tests for client part of QuestionsController.

### CODE REVIEW

File: app/controllers/admins/questions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Admins
+  class QuestionsController < AdminsController
+    before_action :question, only: %i[edit update destroy show]
```
Comment (mmrshk) [inline app/controllers/admins/questions_controller.rb:5]: empty line

File: app/controllers/questions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class QuestionsController < ApplicationController
+  before_action :set_question!
```
Comment (mmrshk) [inline app/controllers/questions_controller.rb:4]: empty line

File: app/controllers/questions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  def index
+    @questions = Question.all
+  end
+
+  def new
+    @question = Question.new
+  end
+
+  def create
+    @question = Question.new(question_params)
```
Comment (mmrshk) [inline app/controllers/questions_controller.rb:14]: empty line

File: app/controllers/questions_controller.rb (hunk 2 above)
Comment (mmrshk) [reply app/controllers/questions_controller.rb:4]: we use bang at the end of the method if this method does destructive changes to DB, this is not that situation

File: app/views/admins/questions/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<h1><%= @question.title %></h1>
+
+<p><%= @question.body %></p>
+<p><%= @question.user_name %></p>
+<p><%= @question.user_email %></p>
+
+
```
Comment (mmrshk) [inline app/views/admins/questions/show.html.erb:7]: empty lines

File: app/views/questions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<h1>New question</h1>
+<%= @question.errors.full_messages %>
+    
+<%= simple_form_for @question do |form| %>
+  <%= form.input :title, label: 'Title' %>
+  <%= form.input :body, label: 'Body' %>
+  <%= form.input :user_name, label: 'Name' %>
+  <%= form.input :user_email, label: 'Email' %>
+  <%= form.button :submit, class: "btn-primary"%>
+<% end %>
```
Comment (mmrshk) [inline app/views/questions/new.html.erb:10]: end of a line

File: app/controllers/admins/questions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    def show
+      @answer = @question.answers.build
+    end
+
+    def edit; end
+
+    def update
+      unless current_user.role_super_admin? || current_user.role_admin?
```
Comment (mmrshk) [inline app/controllers/admins/questions_controller.rb:17]: you know, I think let's remove this, we won't need this code in future

File: app/controllers/questions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show; end
+
+  private
+
+  def set_question!
+    @question = Question.find_by(id: params[:id])
```
Comment (mmrshk) [inline app/controllers/questions_controller.rb:27]: let's use find method in order to have consistency between contollers

File: spec/requests/questions_spec.rb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe 'GET /uk/questions#show' do
+    let(:valid_question) { create(:question) }
+
+    it 'renders a successful response' do
+      get question_url(valid_question, locale: 'uk')
```
Comment (mmrshk) [inline spec/requests/questions_spec.rb:18]: you can use -` i18n.locale `for 'uk'

File: spec/requests/questions_spec.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      end
+
+      it 'creates a new instance of Question with correct values' do
+        expect { post questions_url, params: { question: valid_question } }.to change(Question, :count).by(1)
+      end
+    end
+
+    context 'with invalid parameters' do
+      let(:invalid_question) { build(:question, :question_with_invalid_email) }
+      let(:invalid_params) { attributes_for(:question, :question_with_invalid_email) }
```
Comment (mmrshk) [inline spec/requests/questions_spec.rb:53]: empty line after let

File: app/views/admins/questions/edit.html.erb [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (Bilanuk) [inline app/views/admins/questions/edit.html.erb:1]: delete empty file

**Review by Bilanuk — APPROVED** [review]
Remove empty files if not needed.

File: app/controllers/admins/questions_controller.rb [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    def update
+      if @question.update(question_params)
+        redirect_to admins_questions_path, notice: 'Question updated!'
+      else
+        render :edit, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @question.destroy
```
Comment (mmrshk) [inline app/controllers/admins/questions_controller.rb:24]: you haven't fixed this

**Review by mmrshk — APPROVED** [review]
lgtm

**Pull request merged**